build.gradle

```diff
@@ -1,46 +1,202 @@
+                def ossPass = System.getenv("SONATYPE_PASSWORD") ?: project.hasProperty("sonatypeOssPassword") ? project.sonatypeOssPassword : ''
+                def ossStagingProfileId = System.getenv("SONATYPE_STAGING_PROFILE_ID") ?: project.hasProperty("sonatypeOssStagingProfileId") ? project.sonatypeOssStagingProfileId : ''
+                nexusUrl = uri("https://s01.oss.sonatype.org/service/local/")
+                snapshotRepositoryUrl = uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
+                username = ossUser
+                password = ossPass
+                stagingProfileId = ossStagingProfileId
+            }
+        }
+    }
+}
+
+publishing {
+    if (isSnapshot) {
+        repositories {
+            maven {
+                credentials {
+                    def u = System.getenv("ARTIFACTORY_USERNAME") ?: project.findProperty("artifactoryPublishUsername") ?: ''
+                    def p = System.getenv("ARTIFACTORY_PASSWORD") ?: project.findProperty("artifactoryPublishPassword") ?: ''
+                    username = u
+                    password = p
+                }
+                url = uri('https://repo.grails.org/grails/plugins3-snapshots-local')
+            }
+        }
+    }
+
+    publications {
+        maven(MavenPublication) {
+
+            groupId = project.group
+            version = project.version
 
-    implementation libs.grails.bootstrap
-    implementation libs.grails.core
-    implementation libs.groovy.sql
-    implementation libs.spring.beans
-    implementation libs.spring.context
-    implementation libs.spring.context.support
+            from components.java
 
-    compileOnly libs.groovy.core // CompileStatic and Slf4j
-    compileOnly libs.javax.annotation.api // Provided
+            artifact sourcesJar
+            artifact javadocJar
 
-    testImplementation libs.spock.core
+            pom.withXml {
+
+                def pomNode = asNode()
+                pomNode.children().last() + pomInfo
+
+                // dependency management shouldn't be included
+                try { pomNode.dependencyManagement.replaceNode({}) } catch (Throwable ignore) {}
+            }
+        }
+    }
+}
 
-    testRuntimeOnly libs.slf4j.nop // Get rid of warning about missing slf4j implementation during test task
+if (isReleaseVersion) {
+    afterEvaluate {
+        signing {
+            required = { isReleaseVersion && gradle.taskGraph.hasTask('publish') }
+            sign(publishing.publications.maven)
+        }
+    }
 }
 
-apply from: layout.projectDirectory.file('gradle/java-config.gradle')
-apply from: layout.projectDirectory.file('gradle/grails-plugin-config.gradle')
-apply from: layout.projectDirectory.file('gradle/testing-config.gradle')
-apply from: layout.projectDirectory.file('gradle/documentation-config.gradle')
-apply from: layout.projectDirectory.file('gradle/publishing.gradle')
\ No newline at end of file
+tasks.withType(Sign) {
```

Comment:
There is a lot of code related to publishing in this file.  I'd like to move it to `grailsPublish`.  You already flagged your branch so it can be modified by a maintainer.  I'll make a pass at this today or tomorrow.

Reply:
I just made a quick pass at this, but feel free to take a look.   I think the developer section is probably not complete and I am not sure if there is a way to generate that from the commit log

Reply:
You did great!  When converting to grailsPublish, we've tried to include at least the same developers that were included in the original publishing logic.  Spot checking your entries, it seems you did that.